python/src/Sample_doc.i.in

```diff
@@ -81,6 +81,33 @@ experiments
 1 : [ 2 3 ]
 2 : [ 4 5 ]"
 
+
+%feature("docstring") OT::Sample::BuildFromSequence
+"Static method for building a sample from a collection of double.
+
+Parameters
+----------
+data : 1d array-like
+    Data.
+size : int
+    Size of the sample.
+dimension : int
+    Dimension of the sample.
+
+Returns
+-------
+sample : :class:`~openturns.Sample`
+    Sample with shape and data provided.
+
+
+Examples
+--------
+>>> import openturns as ot
+>>> n = 20
+>>> x = ot.Sample_BuildFromSequence(range(n), n, 1)
+>>> data = [2.0, 2.0, 1.0, 1.0, 2.0, 3.0, 1.0, 2.0, 2.0, 1.0]
+>>> sample = ot.Sample.BuildFromSequence(data, 10, 1)"
```

Comment:
So we cannot simply write `ot.Sample(data, 10, 1)` (or preferably in my opinion `ot.Sample(data, 1)`)?

Reply:
Yet another circular discussion: such a constructor was introduced in the Python layer some times ago, but it was error-prone as it was very close to the Sample(size, point) constructor but with a totally different meaning, hence we removed it. And you propose to reintroduce it.

Here we intend to access the inner storage of a Sample. If you look at SampleImplementation, you see a setData() method, as well as a getData() method (the names were chosen in order to insist on the storage nature of the point, not the elements of the Sample). Only the getData() method has been exposed in the Sample interface, under the name of asPoint(), and is callable only for 1D sample following Denis Barbier's advice. Do you want to introduce a fromPoint() method? Do you want to relax the 1D constraint? If the capacity of the Sample does not match the dimension of the point, I can see these options:
+ throw an exception (the current behavior of setData())
+ if the dimension of the point is greater than the capacity of the sample, keep only the first values and throw the others away silently
+ if the dimension of the point is less than the capacity of the sample, either fill the missing values with zero (this way we match the behavior of std::vector<>.resize()) or recycle the values in a circular way (I can't remember where I saw such a choice. Maybe R?)

But don't reintroduce something we removed after discussion ;-)!

Reply:
There was no discussion on this topic actually.